### PR TITLE
Share 2023 alt CDUs

### PR DESCRIPTION
**Pull request by wrridgeway**
See https://github.com/ccao-data/public/issues/13

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,4 @@
 *.csv
 *.Rda
 *.parquet
+data-requests/pdr-issue-0013/renv/activate.R
```
Comment (dfsnow) [inline .gitignore:27]:
```suggestion
**/renv/activate.R
```

File: data-requests/pdr-issue-0013/pdr-issue-0013.R [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+library(aws.s3)
+library(DBI)
+library(dplyr)
+library(noctua)
+library(openxlsx)
+library(purrr)
+library(readr)
+
+# Local output path
+out_path <- paste0("pdr-issue-0013.xlsx")
```
Comment (dfsnow) [inline data-requests/pdr-issue-0013/pdr-issue-0013.R:11]:
```suggestion
out_path <- "pdr-issue-0013.xlsx"
```

File: data-requests/pdr-issue-0013/pdr-issue-0013.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    par.taxyr,
+    par.class,
+    incentive_classes.user16
+FROM iasworld.pardat AS par
+LEFT JOIN incentive_classes
+    ON par.parid = incentive_classes.parid
+    AND par.taxyr = incentive_classes.taxyr
+WHERE par.deactivat IS NULL
+    AND par.cur = 'Y'
+    AND incentive_classes.user16 IS NOT NULL
```
Comment (dfsnow) [inline data-requests/pdr-issue-0013/pdr-issue-0013.sql:37]:
```suggestion
    AND par.taxyr = incentive_classes.taxyr
    AND incentive_classes.user16 IS NOT NULL
WHERE par.deactivat IS NULL
    AND par.cur = 'Y'
```

File: data-requests/pdr-issue-0013/pdr-issue-0013.sql [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        user16
+    FROM iasworld.oby
+    WHERE deactivat IS NULL AND cur = 'Y'
+    UNION
+    SELECT
+        parid,
+        taxyr,
+        user16
+    FROM iasworld.dweldat
+    WHERE deactivat IS NULL AND cur = 'Y'
```
Comment (dfsnow) [inline data-requests/pdr-issue-0013/pdr-issue-0013.sql:22]: **question (blocking):** Won't each of these return multiple card/lline records and thus not be distinct by PIN/year?

**Review by dfsnow — CHANGES_REQUESTED** [review]
@wrridgeway I don't think this output is what #13 is looking for. The vast majority of alt CDU values in the output don't confer any incentive i.e. (AV, GD, GR). We should probably filter those out.

File: data-requests/pdr-issue-0013/pdr-issue-0013.sql (hunk 4 above)
Comment (wrridgeway) [reply data-requests/pdr-issue-0013/pdr-issue-0013.sql:22]: I think that's fine - we probably don't want to exclude any card that could be an incentive class part of a parcel, even if the primary PIN isn't incentive class.

File: data-requests/pdr-issue-0013/pdr-issue-0013.sql (hunk 3 above)
Comment (wrridgeway) [reply data-requests/pdr-issue-0013/pdr-issue-0013.sql:37]: The goal with joining this way was to exclude pins with non-null from the output, but just from the join... does that make sense?

**Review by dfsnow — APPROVED** [review]

**Pull request merged**